src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -3096,15 +3096,19 @@ namespace FanCoilUnits {
                                                    PLR,
                                                    CompressorOnFlag);
                     }
-                }
-                Calc4PipeFanCoil(state, FanCoilNum, ControlledZoneNum, FirstHVACIteration, QUnitOut);
+                } else if ((CoolingLoad && QUnitOutMax > QZnReq && QZnReq < 0.0) || (HeatingLoad && QUnitOutMax < QZnReq && QZnReq > 0.0)) {
+                    // load is larger than capacity, thus run the fancoil unit at full capacity
+                    PLR = 1.0;
+                } 
+                Calc4PipeFanCoil(state, FanCoilNum, ControlledZoneNum, FirstHVACIteration, QUnitOut, PLR);
                 PowerMet = QUnitOut;
```

Comment:
Added *else if* logic to capture a condition when the heating or cooling load is larger than the maximum equipment capacity and sets the *PLR* to 1.0 since we calling Cal4PipeFanCoil() function with the optional argument *PLR*.